src/db.js

```diff
@@ -0,0 +1,29 @@
+const mongoose = require('mongoose');
+
+class DbConnection {
+  constructor(url) {
+    this.url = url;
+    this.connection = null;
+  }
+
+  async connect() {
+    try {
+      await mongoose.connect(this.url);
+      console.log(`Successfuly connected to ${this.url}`);
+    } catch (err) {
+      console.log(
+        `Connection to ${this.url} has failed. With error code: \n${err}`
+      );
+    }
+  }
+
+  getConnection() {
+    if (this.connection == null) {
+      this.connect();
+      this.connection = mongoose.connection;
+    }
+    return this.connection;
+  }
+}
```

Comment:
If you intend to use the singleton pattern here, make sure that in other files like app.js you're calling the `getConnection()` function and not the `connect()` function otherwise every call to `connect()` will open a new connection which beats the purpose of singleton.